src/ContractInteract/EIP20CoGateway.js

```diff
@@ -385,6 +410,127 @@ class EIP20CoGateway {
     return this.contract.methods.getOutboxMessageStatus(messageHash).call();
   }
 
+  /**
+   * Performs redeem.
+   *
+   * @param {string} amount Amount to redeem.
+   * @param {string} beneficiary Beneficiary address.
+   * @param {string} gasPrice Gas price that staker is willing to pay for the reward.
+   * @param {string} gasLimit Maximum gas limit for reward calculation.
+   * @param {string} nonce Redeemer's nonce.
+   * @param {string} hashLock Hash lock.
+   * @param {Object} txOptions Transaction options.
+   *
+   * @returns {Object} Raw transaction object.
+   */
+  redeem(amount, beneficiary, gasPrice, gasLimit, nonce, hashLock, txOptions) {
+    if (!txOptions) {
+      const err = new TypeError('Invalid transaction options.');
+      return Promise.reject(err);
+    }
+    if (!Web3.utils.isAddress(txOptions.from)) {
+      const err = new TypeError('Invalid redeemer address.');
+      return Promise.reject(err);
+    }
+
+    return this._redeemRawTx(
+      amount,
+      beneficiary,
+      gasPrice,
+      gasLimit,
+      nonce,
+      hashLock,
+    ).then((tx) => Utils.sendTransaction(tx, txOptions));
+  }
+
+  /**
+   * Get the raw transaction for redeem.
+   *
+   * @param {string} amount Amount to redeem.
+   * @param {string} beneficiary Beneficiary address.
+   * @param {string} gasPrice Gas price that staker is willing to pay for the reward.
+   * @param {string} gasLimit Maximum gas limit for reward calculation.
+   * @param {string} nonce Redeemer's nonce.
+   * @param {string} hashLock Hash lock.
+   *
+   * @returns {Object} Raw transaction object.
+   */
+  _redeemRawTx(amount, beneficiary, gasPrice, gasLimit, nonce, hashLock) {
+    if (new BN(amount).eqn(0)) {
+      const err = new TypeError('Redeem amount must not be zero.');
+      return Promise.reject(err);
+    }
+    if (!Web3.utils.isAddress(beneficiary)) {
+      const err = new TypeError('Invalid beneficiary address.');
+      return Promise.reject(err);
+    }
+    if (gasPrice === undefined) {
+      const err = new TypeError('Invalid gas price.');
+      return Promise.reject(err);
+    }
+    if (gasLimit === undefined) {
+      const err = new TypeError('Invalid gas limit.');
+      return Promise.reject(err);
+    }
+    const tx = this.contract.methods.redeem(
+      amount,
+      beneficiary,
+      gasPrice,
+      gasLimit,
+      nonce,
+      hashLock,
+    );
+    return Promise.resolve(tx);
+  }
+
+  /**
+   * Check if the account has approved CoGateway contract for redeem amount transfer.
+   *
+   * @param {string} redmeer Redeemer account address.
+   * @param {string} amount Approval amount.
+   *
+   * @returns {Promise} Promise object.
```

Comment:
Boolean? Maybe say what it returns?
```suggestion
   * @returns {Promise<boolean>} Promise that resolves to `true` if the amount is approved.
```

I don't know what the best practices are here. @hobofan / @sarvesh-ost ?